connect-widget/app/src/shared/api.ts

```diff
@@ -31,17 +31,31 @@ export function getApiToken(searchParams: URLSearchParams): string {
   return apiToken;
 }
 
-export function handleToken(token: string | null): void {
-  if (!token) {
-    isDemo = true;
-  } else if (isDemoToken(token)) {
+export function handleToken(token: string): void {
+  if (isDemoToken(token)) {
     // tell the user widget in demo mode && disable connect buttons
-    isDemo = true;
     throw new DemoTokenError();
-  } else {
-    // invalid vs valid token logic will go here
-    isDemo = false;
   }
+  if (!isTokenValid()) {
+    // tell the user token is invalid and && disable connect buttons
+    throw new InvalidTokenError();
+  }
+
+  // token is present and valid. Set isDemo to false
+  isDemo = false;
+}
+
+function isTokenValid() {
+  try {
+    getApi().get("/connect/redirect", {
+      params: { provider: "fitbit" },
+    });
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (err: any) {
```

Comment:
nit: not need to set the type for `err`, so we can also remove the eslint directive

Reply:
Fixed!